components/server/src/server2.rs

```diff
@@ -1539,14 +1542,18 @@ impl<EK: KvEngine, ER: RaftEngine> EngineMetricsManager<EK, ER> {
 
     pub fn flush(&mut self, now: Instant) {
         let mut reporter = EK::StatisticsReporter::new("kv");
+        let mut collected = false;
         self.tablet_registry
             .for_each_opened_tablet(|_, db: &mut CachedTablet<EK>| {
                 if let Some(db) = db.latest() {
                     reporter.collect(db);
+                    collected = true;
                 }
                 true
             });
-        reporter.flush();
+        if collected {
```

Comment:
Why? Flushing empty should be a no-op.

Reply:
Because arrays are initialized in `collect`. If collect has never been called, flush will panic.